File: solutions/2022. Convert 1D Array Into 2D Array/2022.py
class Solution:
  def construct2DArray(self, original: list[int],
                       m: int, n: int) -> list[list[int]]:
    if len(original) != m * n:
      return []

    ans = [[0] * n for _ in range(m)]

    for i, num in enumerate(original):
      ans[i // n][i % n] = num

    return ans

File: solutions/2022. Convert 1D Array Into 2D Array/2022.java
class Solution {
  public int[][] construct2DArray(int[] original, int m, int n) {
    if (original.length != m * n)
      return new int[][] {};

    int[][] ans = new int[m][n];

    for (int i = 0; i < original.length; ++i)
      ans[i / n][i % n] = original[i];

    return ans;
  }
}
